prow/cmd/deck/static/common/abort.ts

```diff
@@ -0,0 +1,40 @@
+import {showAlert, showToast, State} from "./common";
+
+export function createAbortProwJobIcon(state: string, prowjob: string, csrfToken: string): HTMLElement {
+  const url = `${location.protocol}//${location.host}/abort?prowjob=${prowjob}`;
+  const button = document.createElement('button');
+  button.classList.add('mdl-button', 'mdl-js-button', 'mdl-button--icon');
+  button.innerHTML = '<i class="icon-button material-icons" title="Cancel this job" style="color: gray">cancel</i>';
+
+  if (state !== State.TRIGGERED && state !== State.PENDING) {
+    button.innerHTML = `<i class="icon-button material-icons" title="Can't abort job in ${state} state" style="color: lightgray">cancel</i>`;
```

Comment:
I'd be _very_ careful adding anything into literal HTML that was passed as a simple string via method parameter.

I like to ensure my code works/is safe/etc when viewed "in isolation". Meaning if I _only_ review this one method, I can't ensure that the `state` being passed in isn't some user-provided string.

I think one solution would be to change `string` type to the `enum` type that I think you're _actually_ using somewhere... then I'd know that it's only one of _N_ possibilities and I can navigate directly to those from my IDE. :)

Reply:
This has been updated! It uses another type, `ProwJobState`, that is defined in the prow api (typescript).